### PR TITLE
Enable Prometheus collector to collect data using gRPC endpoints.

### PR DESCRIPTION
**Pull request by vim345**

### CODE REVIEW

File: src/fullerite/collector/prometheus.go [hunk 1]
```diff
@@ -176,12 +186,42 @@ func (p *Prometheus) Collect() {
...
+}
+
+func (p *Prometheus) getGrpcMetrics(endpoint *Endpoint) ([]byte, string, error) {
+	contentType := "application/json"
+	conn, err := grpc.Dial(endpoint.url, grpc.WithInsecure())
+	if err != nil {
+		p.log.Errorf("Failed to connect to server: %s", err)
+		return nil, contentType, err
+	}
+	client := grpcMetrics.NewMetricsClient(conn)
```
Comment (vkhromov) [inline src/fullerite/collector/prometheus.go]: Could you please double-check that this code isn't leaking connections.

File: src/fullerite/collector/prometheus.go [hunk 2]
```diff
@@ -176,12 +186,42 @@ func (p *Prometheus) Collect() {
...
+		return body, contentType, nil
+	}
+}
+
+func (p *Prometheus) getGrpcMetrics(endpoint *Endpoint) ([]byte, string, error) {
+	contentType := "application/json"
+	conn, err := grpc.Dial(endpoint.url, grpc.WithInsecure())
+	if err != nil {
+		p.log.Errorf("Failed to connect to server: %s", err)
+		return nil, contentType, err
```
Comment (vkhromov) [inline src/fullerite/collector/prometheus.go]: `return nil, nil, err`?

File: src/fullerite/collector/prometheus.go [hunk 3]
```diff
@@ -176,12 +186,42 @@ func (p *Prometheus) Collect() {
...
+	}
+	client := grpcMetrics.NewMetricsClient(conn)
+	res, err := client.Metrics(context.Background(), &grpcMetrics.MetricsRequest{})
+	if err != nil {
+		p.log.Errorf("Failed to get the results: %s", err)
+		return nil, contentType, err
```
Comment (vkhromov) [inline src/fullerite/collector/prometheus.go]: `return nil, nil, err`?

File: src/fullerite/collector/prometheus.go [hunk 4]
```diff
@@ -176,12 +186,42 @@ func (p *Prometheus) Collect() {
...
+	if endpoint.isGrpc {
+		return p.getGrpcMetrics(endpoint)
+	} else {
+		body, contentType, scrapeErr := endpoint.httpGetter.Get(
+			endpoint.url,
+			endpoint.headers,
+		)
+		if scrapeErr != nil {
+			p.log.Errorf("Error while scraping %s: %s", endpoint.url, scrapeErr)
+			return nil, contentType, scrapeErr
```
Comment (vkhromov) [inline src/fullerite/collector/prometheus.go]: `return nil, nil, scrapeErr`?

File: src/fullerite/collector/prometheus.go [hunk 5]
```diff
@@ -176,12 +186,42 @@ func (p *Prometheus) Collect() {
...
+		}
+		return body, contentType, nil
+	}
+}
+
+func (p *Prometheus) getGrpcMetrics(endpoint *Endpoint) ([]byte, string, error) {
+	contentType := "application/json"
+	conn, err := grpc.Dial(endpoint.url, grpc.WithInsecure())
```
Comment (vkhromov) [inline src/fullerite/collector/prometheus.go]: Could you please make gRPC respects `timeout` config option?

File: src/fullerite/collector/prometheus.go [hunk 6]
```diff
@@ -109,6 +114,10 @@ func (p *Prometheus) Configure(configMap map[string]interface{}) {
 		if err != nil {
 			p.log.Fatalf("Error while creating HTTP getter: %+v", err)
 		}
+		isGrpc := false
+		if v, exists := endpoint["isGrpc"]; exists {
+			isGrpc = v.(bool)
```
Comment (vkhromov) [inline src/fullerite/collector/prometheus.go]: If `isGrpc == true` then `httpGetter` shouldn't be initialized above and `httpGetter`/`headers` shouldn't be passed into `Endpoint` below.

File: src/fullerite/collector/prometheus.go (hunk 4 above)
Comment (vim345) [reply src/fullerite/collector/prometheus.go]: It doesn't accept `nil` for strings. I set it to an empty string instead.

File: src/fullerite/collector/prometheus.go (hunk 2 above)
Comment (vim345) [reply src/fullerite/collector/prometheus.go]: save as above.

File: src/fullerite/collector/prometheus.go (hunk 3 above)
Comment (vim345) [reply src/fullerite/collector/prometheus.go]: Same as above.

**Comment by vim345** [comment]
@vkhromov Thank you for taking your time and reviewing this.

I created a GRPC utility interface and managed the connections there. Please let me know if I'm missing anything.

I'm not using `text/plain; version=0.0.4` for the contentType. I'm not too sure if this is the right content type though. Do I need to set it to something else?

**Pull request merged**